extensions/AzureOpenAI/AzureOpenAITextGenerator.cs

```diff
@@ -105,6 +95,8 @@ public sealed class AzureOpenAITextGenerator : ITextGenerator
             default:
                 throw new ConfigurationException($"Azure OpenAI: authentication type '{config.Auth:G}' is not supported");
         }
+
+        this._textTokenizer = textTokenizer ?? GptTokenizerDetector.GetTokenizer(this._isTextModel, this._deployment, this._client, this._log);
```

Comment:
this would cause I/O in the ctor, causing fatal errors and service crash in case of a flaky network. It will also increase cost of running the app, with additional requests and tokens spent even when the tokenizer is not used. If the class is not a singleton, or if the code is deployed as a lambda, this would cause a request every time the code is instantiated, with extra costs.

Detecting the model this way is a nice idea, however the feature should be optional and opt-in, and should be probably outside the scope of the class, to maximize the chance of caching the value and reducing cost.

Reply:
I agree, I'll move to Lazy component so the call would be done only the first time the service is really used.